utils/arg_helper.ml

```diff
@@ -0,0 +1,89 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                OCaml                                   *)
+(*                                                                        *)
+(*                       Pierre Chambart, OCamlPro                        *)
+(*           Mark Shinwell and Leo White, Jane Street Europe              *)
+(*                                                                        *)
+(*   Copyright 2015--2016 OCamlPro SAS                                    *)
+(*   Copyright 2015--2016 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file ../LICENSE.       *)
+(*                                                                        *)
+(**************************************************************************)
+
+let fatal err =
+  prerr_endline err;
+  exit 2
+
+module Make (S : sig
+  module Key : sig
+    type t
+    val of_string : string -> t
+    module Map : Map.S with type key = t
+  end
+
+  module Value : sig
+    type t
+    val of_string : string -> t
+  end
+end) = struct
+  type parsed = {
+    default : S.Value.t;
+    override : S.Value.t S.Key.Map.t;
+  }
+
+  let default v = { default = v; override = S.Key.Map.empty }
+
+  let no_equals value =
+    match String.index value '=' with
+    | exception Not_found -> true
+    | _index -> false
+
+  let parse str ~help_text ~update =
+    let values = Misc.Stdlib.String.split str ~on:',' in
+    let parsed =
+      List.fold_left (fun acc value ->
+          match String.index value '=' with
+          | exception Not_found ->
+            begin match S.Value.of_string value with
+            | value -> { acc with default = value }
+            | exception exn ->
+              fatal (Printf.sprintf "%s: %s" (Printexc.to_string exn) help_text)
+            end
+          | equals ->
+            let key_value_pair = value in
+            let length = String.length key_value_pair in
+            if equals <= 0 || equals >= length - 1 then begin
```

Comment:
In a generic library like this one, why do you forbid the empty string? For the key, it's debatable, but for values, it's an unnecessary restriction.